### PR TITLE
[CUMULUS-3527] Adds support for additional kex algorithms to sftp-client

### PR DESCRIPTION
**Pull request by paulpilone**
**Summary:** Summary of changes

Addresses [CUMULUS-3527: SyncGranule failing for SFTP provider](https://bugs.earthdata.nasa.gov/browse/CUMULUS-3527)

After much discussion around this change it was agreed on to simply add specific support for the algorithm that was not by default supported. Our sftp client uses the [ssh2](https://www.npmjs.com/package/ssh2#api) library for ssh connections. By default the `diffie-hellman-group14-sha1` is not supported.

## Changes

* Adds `diffie-hellman-group14-sha1` to the list of supported kex algorithms in the sftp-client

## PR Checklist

- [x] Update CHANGELOG
- [ ] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

**Comment by paulpilone** [comment]
The initial PR only includes the small code change necessary to support the new algorithm. I wanted to get the PR up because it's small and felt like there might be additional discussion around whether or not we should add all of the non-default ones to this list or scope it to the algo specified in the ticket.

I'm currently working on manually testing this by configuring an ssh server that only supports this algorithm. I did not update any unit tests because it feels like including an ssh server specifically for this one algo is overkill and could lead to creep around whether or not we should be testing all of them. Happy to discuss that further if people feel like we should have an automated test for this.

File: packages/sftp-client/src/index.ts [hunk 1]
```diff
@@ -41,6 +41,13 @@ export class SftpClient {
     this.clientOptions = {
       host: config.host,
       port: get(config, 'port', 22),
+      algorithms: {
+        kex: {
+          append: ['diffie-hellman-group14-sha1'],
```
Comment (Jkovarik) [inline packages/sftp-client/src/index.ts:46]: Is there a reason not to append the rest of the optional algorithms (potential removal causing an error to be thrown, etc)?

Comment (paulpilone) [reply packages/sftp-client/src/index.ts:46]: Nope! I originally scoped it to just the one that's called out in the ticket but that's not a strong opinion. I'll add the rest 👍

**Comment by Jkovarik** [comment]
@paulpilone  this looks like a fine approach, I'd be comfortable approving the change given manual testing was done.    Leaving this up for review for further comment, but let me know once it's ready for approval and I'd be happy to do so.

File: packages/sftp-client/src/index.ts (hunk 1 above)
Comment (hkryeung) [reply packages/sftp-client/src/index.ts:46]: Rather be more restrictive than overly accepting I agree; there should be a reason `diffie-hellman-group14-sha1` was not included by default.

**Review by hkryeung — APPROVED** [review]

**Review by Jkovarik — APPROVED** [review]
With a functional demo done to the team (units are inadvisable here), this looks good.  Nice work.

**Pull request merged**